test/N3Store-test.js

```diff
@@ -275,43 +275,115 @@ describe('Store', () => {
 
     it('should decrease the size', () => {
-      store.size.should.eql(5);
+      store.size.should.eql(4);
     });
 
-    it('should match RDF* and normal quads at the same time', done => {
+    it('should match RDF-star and normal quads at the same time', done => {
       const stream = store.removeMatches(null, 'p1', 'o2');
       stream.on('end', () => {
-        store.size.should.eql(3);
+        store.size.should.eql(2);
         done();
       });
     });
 
     it('should allow matching using a quad', done => {
-      const stream = store.removeMatches(termToId(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1'))));
+      const stream = store.removeMatches(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')));
       stream.on('end', () => {
-        store.size.should.eql(1);
+        store.size.should.eql(0);
+        done();
+      });
+    });
+
+    it('should allow matching using a quad and only match against relevant quads', done => {
+      const s2 = new Store([
+        ...allQuads,
+        new Quad(
+          new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2')),
+          new NamedNode('p1'),
+          new NamedNode('o2')),
+      ]);
+
+      const stream = s2.removeMatches(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')));
+      stream.on('end', () => {
+        s2.size.should.eql(2);
         done();
       });
     });
   });
 
+  // These tests should probably be broken in the future; they are here to serve to use that we should to a mver bump
+  // at such a time
+  describe('A store with quoted quads', () => {
+    let store;
+    beforeEach(() => {
+      store = new Store([
+        new Quad(
+          new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2')),
+          new NamedNode('p1'),
+          new NamedNode('o2')),
+        new Quad(
+        new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2'), new NamedNode('g')),
+        new NamedNode('p1'),
+        new NamedNode('o2')),
+      ]);
+    });
+
+    it('should have the correct size', () => {
+      store.size.should.equal(2);
+    });
+
+    it('should get all quads with shared predicate', () => {
+      store.getQuads(null, new NamedNode('p1'), null).length.should.equal(2);
+    });
+
+    it('should get all quads with shared predicate 2', () => {
+      store.getQuads(
+        new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2')),
+        new NamedNode('p1'),
+        null
+        ).length.should.equal(1);
+      store.getQuads(
+        new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2'), new NamedNode('g')),
+        new NamedNode('p1'),
+        null
+        ).length.should.equal(1);
+      store.getQuads(
+        new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2'), new NamedNode('g2')),
+        new NamedNode('p1'),
+        null
+        ).length.should.equal(0);
```

Comment:
To avoid unnecessary breakage downstream; I have written this so that nested quads with graph terms that are *not* the default graph can still be added to the store (as this was the existing behavior when we were just stringifying quad terms).

This is *never* necessary for rdf-star compliant quoted triples; so we may wish to break this behavior at some point.